lib/msal-node/src/client/ClientCredentialClient.ts

```diff
@@ -58,7 +59,20 @@ export class ClientCredentialClient extends BaseClient {
 
         const cachedAuthenticationResult =
             await this.getCachedAuthenticationResult(request);
+
         if (cachedAuthenticationResult) {
+            // if the token is not expired but must be refreshed; get a new one in the background
+            if (this.serverTelemetryManager?.getCacheOutcome()) {
```

Comment:
Did you mean to check a specific outcome?

Reply:
nit: you can return a tuple (token, needs_refersh) from `getCachedAuthenticationResult` to not rely on telemetry.

Reply:
Instead of a tuple, how about a class variable: lastCacheOutcome ?